test/extended/node/oom_detection.go

```diff
@@ -0,0 +1,45 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	exutil "github.com/openshift/origin/test/extended/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TODO: expand this test to SSH into every node in the cluster and gather kubelet logs in case kubelet doesn't send
+// 		 events to the api-server or `oc adm node-logs`
+var _ = g.Describe("[sig-node][Late]", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLIWithoutNamespace("no-oom-kills")
+
+	g.It("should not have pods that are OOM Killed", func() {
+		kubeClient := oc.AdminKubeClient()
+
+		events, err := kubeClient.CoreV1().Events("").List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			g.Fail(fmt.Sprintf("Unexpected error: %v", err))
+		}
+
+		oomEncounteredEvents := []string{}
+		errorMsg := "System OOM encountered"
+		for _, event := range events.Items {
+			if !strings.Contains(event.Message, errorMsg) {
+				continue
+			}
+			oomEncounteredEvent := fmt.Sprintf("OOM encountered within node %v", event.InvolvedObject.Name)
+			oomEncounteredEvents = append(oomEncounteredEvents, oomEncounteredEvent)
+		}
+		failMessage := fmt.Sprintf("%d OOM events occured", oomEncounteredEvents)
```

Comment:
The count is nice, but having a list of OOMed components here would make routing the resulting bugs easier.

Reply:
I think identifying the component which caused OOMKiller(the greedy one) invocation is benefical but AFAIK we can get that info from journal logs.